### PR TITLE
Added various image transformations

### PR DESCRIPTION
**Pull request by XDRAGON2002**
Added various popular image transformations using the hub.compute decorator and created a sample pipeline to demonstrate the same.
Used PIL to implement these features and random to introduce randomness to the transformations in the pipeline.

### CODE REVIEW

**Comment by davidbuniat** [comment]
nice job @XDRAGON2002! yes docstrings would be helpful and can you please make sure if the formatting of the code is following https://github.com/activeloopai/Hub/blob/main/CONTRIBUTING.md

**Comment by XDRAGON2002** [comment]
@davidbuniat Updated PR with proper docstrings, type declarations and formatting.

**Comment by XDRAGON2002** [comment]
@davidbuniat Updated PR with default dataset arguments.

File: transforming/augment.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        help="path to output dataset (default: ./augmented_dataset)",
+    )
+    args = parser.parse_args()
+
+    ds_input = hub.load(args.input_path)
+    ds_output = hub.like(args.output_path, ds_input)
+    pipeline = hub.compose(
+        [
+            cvt_horizontal_flip(probability=0.4),
+            cvt_crop(crop_locations=-1, probability=0.8),
```
Comment (AbhinavTuli) [inline transforming/augment.py:332]: why is -1 passed? It is causing an error with PIL when you try to do img.crop(-1)

Comment (XDRAGON2002) [reply transforming/augment.py:332]: Oh right, my bad, I had initially coded it to crop the center of the image if -1 was passed which I later changed to the default crop if no crop_locations parameter was supplied, and forgot to remove the parameter from the example pipeline. I'll remove it right away.

Comment (XDRAGON2002) [reply transforming/augment.py]: @AbhinavTuli Updated PR with appropriate change.

**Review by davidbuniat — APPROVED** [review]

**Pull request merged**